app/lib/Plugins/SearchEngine/Elastic8ConfigurationSettings.php

```diff
@@ -31,48 +31,32 @@
  */
 
 /* is ElasticSearch running?  */
+
+use Elastic\Elasticsearch\Exception\AuthenticationException;
+use Elastic\ElasticSearch\Exception\ClientResponseException;
+use Elastic\Elasticsearch\Exception\MissingParameterException;
+use Elastic\Elasticsearch\Exception\ServerResponseException;
+
 define('__CA_ELASTICSEARCH_SETTING_RUNNING__', 5001);
 /* does the index exist? */
-define('__CA_ELASTICSEARCH_SETTING_INDEX_EXISTS__', 5002);
+define('__CA_ELASTICSEARCH_SETTING_INDEXES_EXIST__', 5002);
 require_once(__CA_LIB_DIR__ . '/Datamodel.php');
 require_once(__CA_LIB_DIR__ . '/Configuration.php');
 require_once(__CA_LIB_DIR__ . '/Search/SearchBase.php');
 require_once(__CA_LIB_DIR__ . '/Search/ASearchConfigurationSettings.php');
 require_once(__CA_LIB_DIR__ . '/Plugins/SearchEngine/Elastic8.php');
 
 class Elastic8ConfigurationSettings extends ASearchConfigurationSettings {
-	private Configuration $app_config;
-	private Configuration $search_config;
-	private Configuration $search_indexing_config;
-	private array $setting_names;
-	private array $setting_descriptions;
-	private array $setting_hints;
-	private $elasticsearch_base_url;
-	private $elasticsearch_index_name;
+	private array $setting_names = [];
+	private array $setting_descriptions = [];
+	private array $setting_hints = [];
+
+	private ?WLPlugSearchEngineElastic8 $elastic8 = null;
```

Comment:
I _think_ that because you instantiate this in the constructor this is OK to not be nullable:

```suggestion
	private WLPlugSearchEngineElastic8 $elastic8;
```

Reply:
yep, I had gone down a bit of a rabbit hole and forgot to revert this! thanks